include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -107,49 +62,39 @@ class bi_fm_index
     using sdsl_sigma_type = typename sdsl_index_type::alphabet_type::sigma_type;
 
     //!\brief The type of the underlying FM index for the original text.
-    using fm_index_type = fm_index<text_t, typename index_traits_t::fm_index_traits>;
+    using fm_index_type = fm_index<is_collection, sdsl_index_type>;
 
     //!\brief The type of the underlying FM index for the reversed text.
-    using rev_fm_index_type = fm_index<rev_text_type, typename index_traits_t::rev_fm_index_traits>;
+    using rev_fm_index_type = fm_index<is_collection, sdsl_index_type>;
     //!\}
 
-    //!\brief Access to a reversed view of the text. Needed when a unidirectional cursor on the reversed text is
-    //        constructed from the bidirectional index.
-    rev_text_type rev_text;
-
     //!\brief Underlying FM index for the original text.
     fm_index_type fwd_fm;
 
     //!\brief Underlying FM index for the reversed text.
     rev_fm_index_type rev_fm;
 
 public:
-
     /*!\name Text types
      * \{
      */
-    //!\brief The type of the underlying character of text_type.
-    using char_type = innermost_value_type_t<text_t>;
     //!\brief Type for representing positions in the indexed text.
     using size_type = typename sdsl_index_type::size_type;
     //!\}
 
-    //!\brief The index traits object.
-    using index_traits = index_traits_t;
-
     /*!\name Cursor types
      * \{
      */
     //!\brief The type of the bidirectional cursor.
-    using cursor_type = bi_fm_index_cursor<bi_fm_index<text_t, index_traits_t>>;
+    using cursor_type = bi_fm_index_cursor<bi_fm_index<is_collection, sdsl_index_type>>;
     //!\brief The type of the unidirectional cursor on the original text.
     using fwd_cursor_type = fm_index_cursor<fm_index_type>;
     //!\brief The type of the unidirectional cursor on the reversed text.
     using rev_cursor_type = fm_index_cursor<rev_fm_index_type>;
     //!\}
 
-    template <typename bi_fm_index_t>
-    friend class bi_fm_index_cursor;
+    //!\brief The alphabet size of the text.
+    size_t sigma{0};
```

Comment:
This is public?

Reply:
💡Actually this can be private since the iterators are befriended...